### ISSUE TITLE
no-unused-vars warnings in array destructuring

### ISSUE THREAD

**Issue by jakepoz** [eslint/eslint#12175]
**Tell us about your environment**

* **ESLint Version:** 6.2.2
* **Node Version:** 10.14.1
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**  Babel-ESLint (Create React app)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
  "eslintConfig": {
    "extends": "react-app"
  },
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
for (let [code, step, color, tag] of statusCases) {
            if (code === chapter.status) {
                if (color !== IN_PROGRESS_COLOR)
                    return <div className={`ui ${color} horizontal label`}>{tag}</div>;
                else
                    return <div className={`ui ${color} horizontal label`}>Step {step}/{TOTAL_STEPS} - {tag}</div>;
            }
        }
```

```bash
npm run-script build 
(Create React App)
```

**What did you expect to happen?**
No warnings, as all variables are used

**What actually happened? Please include the actual, raw output from ESLint.**
  Line 100:  'code' is defined but never used   no-unused-vars
  Line 100:  'step' is defined but never used   no-unused-vars
  Line 100:  'color' is defined but never used  no-unused-vars
  Line 100:  'tag' is defined but never used    no-unused-vars

**Are you willing to submit a pull request to fix this bug?**
No
